__tests__/Collection.spec.ts

```diff
@@ -17,34 +23,49 @@ describe(Collection, () => {
   beforeEach(() => {
     collection = new MockCollection([
       { id: 1, phone: '1234' },
-      { id: 2, phone: '5678' }
+      { id: 2, phone: '5678' },
+      { id: 3, phone: null },
+      { id: 4 }
     ])
   })
 
   describe('length', () => {
     it('returns the number of models', () => {
-      expect(collection.length).toBe(2)
-    })
-  })
-
-  describe('url()', () => {
-    it('throws', () => {
-      expect(() => collection.url()).toThrow('You must implement this method')
+      expect(collection.length).toBe(4)
     })
   })
 
   describe('toJS()', () => {
     it('returns a plain representation of the models array', () => {
       expect(collection.toJS()).toEqual([
         { id: 1, phone: '1234' },
-        { id: 2, phone: '5678' }
+        { id: 2, phone: '5678' },
+        { id: 3, phone: null },
+        { id: 4 }
       ])
     })
   })
 
+  describe('index', () => {
+    it('indexes the collection', () => {
+      expect(collection.index.get('id')).toEqual(new Map([
+        [1, collection.filter({ id: 1 })],
+        [2, collection.filter((model) => model.get('id') === 2)],
+        [3, collection.filter({ id: 3 })],
+        [4, collection.filter({ id: 4 })]
+      ]))
+
+      expect(collection.index.get('phone')).toEqual(new Map([
+        ['1234', [collection.find({ phone: '1234' })]],
+        ['5678', [collection.find((model) => model.get('phone') === '5678')]],
```

Comment:
Update the `find` documentation to allow functions. Right now there is only the `Object` way. Following this path, the `mustFind` should accept the `Function` way too.

Reply:
good catch. Documentation needs a lot of tuning.